### DISCUSSION TITLE
Write a package over lefthook, use as a dependency in other projects

### DISCUSSION THREAD

**Discussion by fostimus** [evilmartians/lefthook#585] [Ideas]
My team has many projects that use git hooks. They are always the same hooks, we just copy the logic over.

It would be really beneficial to us if we could write an NPM package over our git hooks manager (currently, husky) and then by installing the package, the project we installed the package to just "gets" the git hooks automatically. Maybe something in lefthook could be exposed to specify a parent directory to run against the .git folder?

For example
```
. project-root
|---- README.md
|----.git
|---- node_modules
     |---- custom-lefthook-pkg
          |---- lefthook.yml
          |---- package.json
```

Is there a way to have `custom-lefthook-pkg` enable it's git hooks on the project-root?

Or does this exist already? Thank you 🙏🏼

**Comment by mrexox** [comment]
Hey! Looks like you need to distribute the lefthook configuration across multiple repos. There's an option for that – [`remote`](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#remote). You'll need to add a git repo with the configuration and then use the following template in every repo that needs it.

```yml
remote:
  git_url: https://github.com/organization/lefthook-config
```

When you install `lefthook` npm package it runs `lefthook install` and setups the git hooks from the remote repo.

This is the currently supported approach. I'll be glad to know more about your use case, so I could help with configuration.
